File: app/src/main/java/com/example/a39092/rxjavatest/module/main/view/LvMovieAdapter.java
package com.example.a39092.rxjavatest.module.main.view;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.example.a39092.rxjavatest.R;
import com.example.a39092.rxjavatest.model.bean.ResponseBody;
import com.jakewharton.rxbinding2.view.RxView;

import java.util.List;
import java.util.concurrent.TimeUnit;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.functions.Consumer;

/**
 * @author chenxi
 * @title RxJavaTest
 * @date 2017/4/13
 * @description
 */

public class LvMovieAdapter extends BaseAdapter {
    private List<ResponseBody.SubjectsBean> list;
    private Context context;

    public LvMovieAdapter(List<ResponseBody.SubjectsBean> list, Context context) {
        this.list = list;
        this.context = context;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = View.inflate(context, R.layout.lv_movie_item, null);
            holder = new ViewHolder(convertView);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }
        holder.lvMovieItemName.setText(list.get(position).getTitle());
        holder.lvMovieItemStar.setText(list.get(position).getRating().getStars());
        Glide.with(context).load(list.get(position).getImages().getMedium()).into(holder.lvMovieItemPic);
        RxView.clicks(convertView).throttleFirst(1, TimeUnit.SECONDS).subscribe(new Consumer<Object>() {
            @Override
            public void accept(Object o) throws Exception {
                Toast.makeText(context, position, Toast.LENGTH_SHORT).show();
            }
        });
        return convertView;
    }

    class ViewHolder {
        @BindView(R.id.lv_movie_item_pic)
        ImageView lvMovieItemPic;
        @BindView(R.id.lv_movie_item_name)
        TextView lvMovieItemName;
        @BindView(R.id.lv_movie_item_star)
        TextView lvMovieItemStar;

        ViewHolder(View view) {
            ButterKnife.bind(this, view);
        }
    }
}

File: app/src/main/java/com/example/a39092/rxjavatest/module/main/view/LvMovieAdapter.kt
package com.example.a39092.rxjavatest.module.main.view

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import com.bumptech.glide.Glide
import com.example.a39092.rxjavatest.R
import com.example.a39092.rxjavatest.model.bean.ResponseBody
import com.jakewharton.rxbinding2.view.RxView
import java.util.concurrent.TimeUnit

/**
 * @author chenxi
 * *
 * @title RxJavaTest
 * *
 * @date 2017/4/13
 * *
 * @description
 */

class LvMovieAdapter(private val list: ArrayList<ResponseBody.SubjectsBean>, private val context: Context) : BaseAdapter() {

    override fun getCount(): Int {
        return list.size
    }

    override fun getItem(position: Int): Any {
        return list[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val holder: ViewHolder
        if (convertView == null) {
            convertView = View.inflate(context, R.layout.lv_movie_item, null)
            holder = ViewHolder(convertView)
            convertView!!.tag = holder
        } else {
            holder = convertView.tag as ViewHolder
        }
        holder.lvMovieItemName!!.text = list[position].title
        holder.lvMovieItemStar!!.text = list[position].rating!!.stars
        Glide.with(context).load(list[position].images!!.medium).into(holder.lvMovieItemPic!!)
        RxView.clicks(convertView).throttleFirst(1, TimeUnit.SECONDS).subscribe { Toast.makeText(context, position.toString(), Toast.LENGTH_SHORT).show() }
        return convertView
    }

    internal inner class ViewHolder(view: View) {
        var lvMovieItemPic: ImageView? = null
        var lvMovieItemName: TextView? = null
        var lvMovieItemStar: TextView? = null

        init {
            lvMovieItemPic = view.findViewById(R.id.lv_movie_item_pic) as ImageView?
            lvMovieItemName = view.findViewById(R.id.lv_movie_item_name) as TextView?
            lvMovieItemStar = view.findViewById(R.id.lv_movie_item_star) as TextView?
        }
    }
}
